modules/Microsoft.Storage/static-website/1.0/main.bicep

```diff
@@ -0,0 +1,91 @@
+@description('The location into which the Azure Storage resources should be deployed.')
+param location string = resourceGroup().location
+
+@description('The name of the Azure Storage account to create. This must be globally unique.')
+param accountName string = 'stor${uniqueString(resourceGroup().id)}'
+
+@description('The name of the SKU to use when creating the Azure Storage account.')
+@allowed([
+  'Standard_LRS'
+  'Standard_GRS'
+  'Standard_ZRS'
+  'Standard_GZRS'
+  'Premium_LRS'
+])
+param skuName string = 'Standard_LRS'
+
+@description('The name of the page to display when a user navigates to the root of your static website.')
+param indexDocument string = 'index.htm'
+
+@description('The name of the page to display when a user attempts to navigate to a page that does not exist in your static website.')
+param errorDocument404Path string = '404.htm'
+
+@description('Indicates whether the storage account should require HTTPS traffic.')
+param supportsHttpsTrafficOnly bool = true
+
+var storageAccountContributorRoleDefinitionId = subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '17d1049b-9a84-46fb-8f53-869881c3d3ab') // as per https://docs.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#:~:text=17d1049b-9a84-46fb-8f53-869881c3d3ab
+var storageAccountStorageBlobDataContributorRoleDefinitionId = subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'ba92f5b4-2d11-453d-a403-e96b0029c9fe') // as per https://docs.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#:~:text=ba92f5b4-2d11-453d-a403-e96b0029c9fe
+var managedIdentityName = 'StorageStaticWebsiteEnabler'
+var deploymentScriptName = 'EnableStorageStaticWebsite'
+
+resource storageAccount 'Microsoft.Storage/storageAccounts@2021-04-01' = {
+  name: accountName
+  location: location
+  kind: 'StorageV2'
+  sku: {
+    name: skuName
+  }
+  properties: {
+    supportsHttpsTrafficOnly: supportsHttpsTrafficOnly
+  }
+}
+
+resource managedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = {
+  name: managedIdentityName
+  location: location
+}
+
+resource roleAssignmentContributor 'Microsoft.Authorization/roleAssignments@2020-04-01-preview' = {
+  scope: storageAccount
+  name: guid(resourceGroup().id, managedIdentity.id, storageAccountContributorRoleDefinitionId)

```

Comment:
since this has the scope of the storageAccount, the guid should as well, e.g.

guid(storageAccount.id, managedIdentity.id, storageAccountContributorRoleDefinitionId)

Reply:
Good catch - fixed.